src/attendees/attendee.py

```diff
@@ -60,26 +100,114 @@ def has_attribute(self, key: Hashable) -> bool:
+
+    @property
+    def has_file_url(self):    
+        return self._file_url == None
+
+    @property
+    def has_file_path(self):    
+        return self._file_url == None
+    
     def set_attribute(self, key: Hashable, value: Any,
-                     verifyFunc: VerifyFunc = None):
+                     verifyFunc: VerifyFunc = None): #verifyFunc is a function, a callback - the what, for miscellaneous, 
+                    # i wont write callback, will use it
         """
         Set an arbitrary attribute of the attendee.
 
-        Will add the attribute if it doesn't exist.
+        Will add the attribute if it doesn't exist. First Checks if `verifyFunc`
+        is `None`. If `None`, do not verify value, any value is validOnce verified,
+        set misc attribute using `key`.
 
         Args:
             key (Hashable): The name of the attribute.
             value (Any): The attribute value.
             verifyFunc (Callable[[Hashable, Any], bool]): A function to verify
-            the value being set, if let as None, will not verify value.
+            the value being set, if let as `None`, will not verify value. If
+            `True`, value is valid and vice-versa.
         
         Raises:
-            TypeError if key is not hashable.
-            ValueError if key is "fname", "lname", or "email"
+            TypeError: if `key` is not hashable.
+            ValueError: if `key` is "fname", "lname", or "email"
+            ValueError: if verifyFunc returns `False`
         """
-        raise NotImplementedError
+        if verifyFunc is not None: # if None, dont need to verify
+            verifyFunc(key, value) # need to make sure its correct
+            if not verifyFunc(key, value):
+                # raise error - value is not valid fo this key
+                raise NotImplementedError
+
+        self._misc[key] = value
+
+    @classmethod
+    def checkString(cls, value):    # checks if string and if empty
+        # cls, allows u to use func without creating object. MyClass.func()
+        if type(value) != str:
+            raise TypeError("Value should be a string")
+        if value == "":
+            raise ValueError("Value is empty")
+
+    @classmethod
+    def verifyName(cls, string): 
+        cls.checkString(string)
+        if string[0].islower():    # checks for first letter capital
+            raise ValueError("Name must start with capital letter")
+        
+        n = len(string)-1
+        if string[n] in cls.invalid_name_chars:    # checks if ends in - ' or space
+            raise ValueError(f"Name cannot end with {string[n]}")
+        
+        # ref a ptr that automatically derefs when using it, all objects refer to w ref
+        # == : checks for equivalence, 2 sep objects, for whats inside it
+        # is : checks for same thing, 2 references pointing to same thing. 
+        # if head is tail for linked list 1 element. head == tail, 'bob' at both tail & head
```

Comment:
Please remove this comments explaining the difference between `==` and `is`. It will make no sense to anyone else reading this code.